apispec/ext/marshmallow/__init__.py

```diff
@@ -82,11 +80,11 @@ def resolve_parameters(self, parameters):
         resolved = []
         for parameter in parameters:
             if not isinstance(parameter.get('schema', {}), dict):
-                schema_cls = resolve_schema_cls(parameter['schema'])
-                if issubclass(schema_cls, marshmallow.Schema) and 'in' in parameter:
+                schema_instance = resolve_schema_instance(parameter['schema'])
+                if 'in' in parameter:
                     del parameter['schema']
                     resolved += self.openapi.schema2parameters(
-                        schema_cls,
+                        schema_instance,
```

Comment:
This change is to prevent a schema like `SelfReferencingSchema` passed as a query string parameter from throwing an error due to unbound nested self referencing fields.  It is not _valid_ to have a self referencing schema in that location, but at least it doesn't throw an error.